WebDevStudios/constant-contact-forms
# CC-36: Allow users to choose list

Ticket: [CC-36](https://webdevstudios.atlassian.net/browse/CC-36)

### Description ###

Updates admin and frontend form display to allow list options to customer:
- Changes admin list selection to multi-checkbox.
- Updates frontend list display:
    - Separates list input from "opt-in" checkbox/text.
    - Displays checkboxes for each list selected in admin.
- Updates contact submission process to handle multiple lists.
- Updates admin forms listing to display potentially multiple lists in list column.

### Screenshot ###

![Screen Shot 2020-07-06 at 4 54 02 PM](https://user-images.githubusercontent.com/36422618/86805042-bf632680-c034-11ea-98b8-341e1417da3d.png)

### Steps to verify the solution ###

1. Edit form in admin and confirm a) list selection now display as a checkbox, and b) previous list selection is automatically checked.
2. In admin forms listing, confirm all forms display appropriate list(s) in the list column.
3. On the frontend, confirm a) edited form displays all list options selected in admin as checkboxes, and b) opt-in checkbox is separate from list selection.
4. Attempt to submit form without selecting a list and confirm error message appears.
5. Submit form with a) one list selected, then b) again with more than one list selected.
6. Edit form in admin to select only a single list.
7. On the frontend, confirm single checkbox is hidden but still exists, then submit form.
6. Log into Constant Contact account and confirm a) submissions were successful, and b) contact was added to all lists selected. (Note: Selecting "Bypass Constant Contact cron scheduling" in Contact Form settings is useful to avoid delays in submission.)
